### ISSUE TITLE
rtl8723bs-dkms_4.14_all.deb no longer installing

### ISSUE THREAD

**Issue by moonbit5** [weese/Circuit-Sword#8]
hey @weese I've encountered an issue where I can no longer install the .deb after updating my system: 

```shell
./fix-for-installed-kernel.sh
kernel = /boot/kernel7.img
offset = 18624 bytes
headers = 5.10.103-v7+
Source code for 5.10 does not exists
Checking for upgrade patch
Patch not found
Trying to download a new sources..
echo "Need a subversion for this!"
Need a subversion for this!
svn export https://github.com/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs 5.10
svn: E170013: Unable to connect to a repository at URL 'https://github.com/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs'
svn: E160013: '/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs' path not found
make: *** [Makefile:32: 5.10] Error 1
``` 
Would really appreciate your knowledge on fixing the missing url in the script. Thank you!

**Comment by moonbit5** [comment]
@Antho91 maybe you might know the answer to this?

**Comment by Antho91** [comment]
Tbh I haven't looked at this for over a year, so I don't know the answer. It's trying to download a package, which is no longer there. Probably the package needs to be updated (or the kernel).

**Comment by moonbit5** [comment]
thanks @Antho91 correct me if I'm wrong but the Makefile has the wrong URL (since it was removed from github) so updating the kernel wouldn't change that? Or are you saying that the subversion package needs updating?

**Comment by Antho91** [comment]
Finally got some time to look into it. I believe you've forgotten to place a file called wpa_supplicant.conf into the root of your SD card ;)

So create a file called 'wpa_supplicant.conf'
In the file plz add:

```
country=US
ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
update_config=1

network={
    ssid="abc"
    psk="123"
}
```
Change the SSID to your Wi-Fi name and of course change the password ^^

**Comment by moonbit5** [comment]
Thanks for looking at this. I'm fairly sure I was connected to the internet, however I will double check

**Comment by djdiskmachine** [comment]
I managed to set my WiFi csid and pw in raspi-config when flashing a new card. When rebooting there was the 'no Wireless interface found'
The setup  had created a supplicant file but without the country field at the top. Setting this does not solve the issue though. 

When running raspi-config again In addition to the no interface error i get a second error 'there was an error running option L4 WLAN country '

**Comment by Antho91** [comment]
Did you by any change run the update command?

If yes, that breaks the wifi and probable the audo dkms, because the urls no longer exist(have been changed). Im not a pro, but was already trying to solve the issue

**Comment by moonbit5** [comment]
The URLs in the script (on this repository) no longer exist and the script fails

**Comment by Antho91** [comment]
Yeah I saw that already👍. I was first fixing a few issues for building a new img, I have that now and I have now an img with the latest kernel.

Next fix, is trying to figure out where the old URL are coming from, changing it in the Makefile is not the solution. Probably I need to decompile the .deb file. But I need time, which I don't have😄

Anyway it's fixable 

But I'm still wondering when do you get this error. Because i don't get that error? Is it right after flashing the img and booting the device? Or somewhere else? That info I'm still missing

**Comment by moonbit5** [comment]
That's great. I was just attempting to update the underlying OS.

Basically I was attempting to compile and install es-de frontend or the latest attract mode plus frontend. But both of them require OpenGL ES 3.0 which is why I was trying to update.

And actually what I had done was install a fresh version of raspberry pi os and I'd cloned this repo to to try and install all the nessesary drivers for the circuit sword

**Comment by djdiskmachine** [comment]
> Did you by any change run the update command?

I did yes, guess I should have added that to my description. ^_^'
Audio works really good tho

**Comment by Antho91** [comment]
Oke did some digging with no result. Somehow, my Makefile gets rewritten with the incorrect URL. I have no clue how and why this happens :(.

So I'm going to create my own dkms files xD. But I don't know when I can release them (#nopressure).

For now, don't update the installed IMG and everything should work ;).

I've also updated my repository so that it will build with kernel version 6.6.

So things to-do:
- Create a new DKMS for the Wi-Fi
- Create a new DKMS for the audio, because this driver also fails after the update procedure
- Remove the need of using downloaded packages (so that you always use the latest packages, while building the IMG)
- Release a version with the updated kernel 6.6. with the new DKMS drivers so that the functionality survives after a kernel update

If you have knowledge or spare time and wants to help out, feel free ;)

**Comment by djdiskmachine** [comment]
> Oke did some digging with no result. Somehow, my Makefile gets rewritten with the incorrect URL. I have no clue how and why this happens :(.

Exactly what URL are you referring to?
Which do you want to set and what is it overwritten with?

**Comment by Antho91** [comment]
OK, some background info (not sure if I'm on the right track).

### TLDR;
I think this is our issue. Regarding the usage of 'Subversion' https://github.blog/news-insights/product-news/sunsetting-subversion-support/

I've unpacked both .deb files and changed it to the new URL's, but even that didn't work. Probably due to subversion (SVN command). 

I've also was trying to create new DKMS .deb packages, but I didn't have any good result. See:  https://github.com/Antho91/Circuit-Sword/tree/wifi-dev (sound-module and WIFI-firmware folder) If we get that working, I think we are up and running. It was complaining about the kernel headers (missing /build folder in the kernel folder). Even though I've installed (and re-installed the 'raspberrypi-kernel-headers').

**So still my answer is do not update :)**

### My investigation 👍

After running:
`sudo apt update && sudo apt full-upgrade -y`
I see in the output the following errors for the DKMS packages:

```
Removing 'diversion of /boot/overlays/wm8960-soundcard.dtbo to /usr/share/rpikernelhack/overlays/wm8960-soundcard.dtbo by rpikernelhack'
run-parts: executing /etc/kernel/postinst.d/apt-auto-removal 5.10.103+ /boot/kernel.img
run-parts: executing /etc/kernel/postinst.d/dkms 5.10.103+ /boot/kernel.img
/etc/kernel/postinst.d/dkms:
Error! Bad return status for module build on kernel: 5.10.103+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/postinst.d/initramfs-tools 5.10.103+ /boot/kernel.img
run-parts: executing /etc/kernel/postinst.d/apt-auto-removal 5.10.103-v7+ /boot/kernel7.img
run-parts: executing /etc/kernel/postinst.d/dkms 5.10.103-v7+ /boot/kernel7.img
/etc/kernel/postinst.d/dkms:
Error! Bad return status for module build on kernel: 5.10.103-v7+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103-v7+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/postinst.d/initramfs-tools 5.10.103-v7+ /boot/kernel7.img
run-parts: executing /etc/kernel/postinst.d/apt-auto-removal 5.10.103-v7l+ /boot/kernel7l.img
run-parts: executing /etc/kernel/postinst.d/dkms 5.10.103-v7l+ /boot/kernel7l.img
/etc/kernel/postinst.d/dkms:
Error! Bad return status for module build on kernel: 5.10.103-v7l+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103-v7l+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/postinst.d/initramfs-tools 5.10.103-v7l+ /boot/kernel7l.img
run-parts: executing /etc/kernel/postinst.d/apt-auto-removal 5.10.103-v8+ /boot/kernel8.img
run-parts: executing /etc/kernel/postinst.d/dkms 5.10.103-v8+ /boot/kernel8.img
/etc/kernel/postinst.d/dkms:
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
run-parts: executing /etc/kernel/postinst.d/initramfs-tools 5.10.103-v8+ /boot/kernel8.img
```



```
Setting up raspberrypi-kernel-headers (1:1.20230509~buster-1) ...
run-parts: executing /etc/kernel/header_postinst.d/dkms 5.10.103+
Error! Bad return status for module build on kernel: 5.10.103+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/header_postinst.d/dkms 5.10.103-v7+
Error! Bad return status for module build on kernel: 5.10.103-v7+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103-v7+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/header_postinst.d/dkms 5.10.103-v7l+
Error! Bad return status for module build on kernel: 5.10.103-v7l+ (armv7l)
Consult /var/lib/dkms/rtl8723bs/4.14/build/make.log for more information.
Error! Bad return status for module build on kernel: 5.10.103-v7l+ (armv7l)
Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log for more information.
run-parts: executing /etc/kernel/header_postinst.d/dkms 5.10.103-v8+
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
Error! echo
Your kernel headers for kernel 5.10.103-v8+ cannot be found at
/lib/modules/5.10.103-v8+/build or /lib/modules/5.10.103-v8+/source.
```


Consulting the following and I see (Consult /var/lib/dkms/snd-usb-audio/0.1/build/make.log)

```
(DKMS make.log for snd-usb-audio-0.1 for kernel 5.10.103-v7l+ (armv7l)
Thu 17 Oct 08:02:31 BST 2024
Source code for 5.10 does not exists
Checking for upgrade patch
Patch not found
Trying to download new sources..
echo "Need a subversion for this!"
Need a subversion for this!
svn export https://github.com/raspberrypi/linux/branches/rpi-5.10.y/sound/usb 5.10
svn: E170013: Unable to connect to a repository at URL 'https://github.com/raspberrypi/linux/branches/rpi-5.10.y/sound/usb'
svn: E670003: Temporary failure in name resolution
make: *** [Makefile:32: 5.10] Error 1)
```

Where we can see that the URL is https://github.com/raspberrypi/linux/branches/rpi-5.10.y/sound/usb which gives a 410 
changing it to https://github.com/raspberrypi/linux/tree/rpi-5.10.y/sound/usb, and we can see the data of that branch  (in the browser).

Same if we look at  /var/lib/dkms/rtl8723bs/4.14/build/make.log

```
DKMS make.log for rtl8723bs-4.14 for kernel 5.10.103-v7l+ (armv7l)
Thu 17 Oct 08:02:31 BST 2024
Source code for 5.10 does not exists
Checking for upgrade patch
Patch not found
Trying to download a new sources..
echo "Need a subversion for this!"
Need a subversion for this!
svn export https://github.com/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs 5.10
svn: E170013: Unable to connect to a repository at URL 'https://github.com/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs'
svn: E670003: Temporary failure in name resolution
make: *** [Makefile:32: 5.10] Error 1
```

We can see that we use the URL https://github.com/raspberrypi/linux/branches/rpi-5.10.y/drivers/staging/rtl8723bs which gives a 410 changing it to https://github.com/raspberrypi/linux/tree/rpi-5.10.y/drivers/staging/rtl8723bs, and it's giving me data (in the browser)

I'm also not sure because we use 'subversion' if this is the whole issue https://github.blog/news-insights/product-news/sunsetting-subversion-support/ 

I've unpacked both .deb files and changed it to the new URL's, but even that didn't work. Probably due to subversion. 

I've also was trying to create new DKMS .deb packages, but I didn't have any good result. See:  https://github.com/Antho91/Circuit-Sword/tree/wifi-dev (sound-module and WIFI-firmware folder) If we get that working, I think we are up and running.

So still my answer is do not update :)

**Comment by djdiskmachine** [comment]
Seems like this dkms might be not needed with modern kernels
I'll try a fix somewhat like this at some point:

`#!/bin/bash

echo "[Circuit Sword] Cleaning up legacy DKMS modules..."

# List of problematic DKMS modules
MODULES=(
  "rtl8723bs/4.14"
  "snd-usb-audio/0.1"
)

for module in "${MODULES[@]}"; do
  if dkms status | grep -q "$module"; then
    echo "Removing DKMS module: $module"
    sudo dkms remove "$module" --all || true
  fi
done

echo "[Circuit Sword] Removing related packages if installed..."

sudo apt purge -y rtl8723bs-dkms snd-usb-audio-dkms || true

echo "[Circuit Sword] Cleaning residual DKMS state..."
sudo rm -rf /var/lib/dkms/rtl8723bs
sudo rm -rf /var/lib/dkms/snd-usb-audio

echo "[Circuit Sword] Done cleaning DKMS modules."`
